odk1-src/form-logic.rst

```diff
@@ -0,0 +1,314 @@
+***********
+Form Logic
+***********
+
+Form logic building blocks
+============================
+
+.. _expressions:
+  
+Expressions
+-------------
+
+An :dfn:`expression`, sometimes called a `formula`_,
+is evaluated dynamically as a form is filled out.
+It can include `XPath functions`_, :ref:`operators`,
+:ref:`values from previous responses <previous-responses>`,
+and (in some cases) :ref:`the value of the current response <current-value>`. 
+
+.. _XPath functions: https://opendatakit.github.io/xforms-spec/#xpath-functions
+
+.. _formula: http://xlsform.org/#formulas
+
+.. rubric:: Example expressions
+
+:tc:`${bill_amount} * 0.18`
+  Multiplies the previous value :tc:`bill_amount` by 18%,
+  to calculate a suitable tip.
+  
+:tc:`concat(${first_name}, ' ', ${last_name})`
+  Concatenates two previous responses with a space between them 
+  into a single string.
+  
+:tc:`${age} >= 18`
+  Evaluates to ``True`` or ``False``, 
+  depending on the value of :tc:`age`.
+
+:tc:`round(${bill_amount} * ${tip_percent} * 0.01, 2)`
+  Calculates a tip amount based on two previously entered values,
+  and then rounds the result to two decimal places.
+
+Expressions are used in:
+
+ - :ref:`calculations`
+ - :ref:`constraints`
+ - :ref:`relevants`
+
+
+
+.. _variables:
+
+Variables
```

Comment:
I think the simplest items should go first. So variables, then expressions, then calculations.